helm/nginx-ingress-controller-app/templates/controller-service-internal.yaml

```diff
@@ -0,0 +1,46 @@
+{{- if .Values.controller.service.internal.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    external-dns.alpha.kubernetes.io/hostname: ingress.{{ .Values.baseDomain }}
```

Comment:
This is the same domain name as the public load balancer.
What happens if both the public and internal load balancers are enabled?
I expect the DNS record to have both IP addresses as values. Is that what we want?
I feel having the same DNS record resolving with a public and a private IP address isn't the right thing to do.

Reply:
Ack. FYI, this is a draft PR, about to test it. Have a proposal when it comes to naming?

Reply:
I'd go with some self-explanatory thing like `ingress-internal`. WDYT?

PS shout if you need help with tests

Reply:
```suggestion
    external-dns.alpha.kubernetes.io/hostname: ingress-internal.{{ .Values.baseDomain }}
```
WDYT?

Reply:
Merged name change.

I'll test AWS node pools. If you can help with Azure that would be great.

Build version to be tested: 1.7.0-b3ca4b88bfa0c39b5e8f58546b810182459b7f17